cupy_backends/cuda/libs/_cnvrtc.pxi

```diff
@@ -0,0 +1,139 @@
+cdef F_nvrtcGetNVVMSize nvrtcGetNVVMSize
+
+ctypedef nvrtcResult (*F_nvrtcGetNVVM)(nvrtcProgram prog, char *nvvm) nogil
+cdef F_nvrtcGetNVVM nvrtcGetNVVM
+
+
+cdef SoftLink _L = None
+cdef void initialize():
+    global _L
+    if _L is not None:
+        return
+    _L = _get_softlink()
+
+    global nvrtcGetErrorString
+    nvrtcGetErrorString = <F_nvrtcGetErrorString>_L.get('GetErrorString')
+    global nvrtcVersion
+    nvrtcVersion = <F_nvrtcVersion>_L.get('Version')
+    global nvrtcCreateProgram
+    nvrtcCreateProgram = <F_nvrtcCreateProgram>_L.get('CreateProgram')
+    global nvrtcDestroyProgram
+    nvrtcDestroyProgram = <F_nvrtcDestroyProgram>_L.get('DestroyProgram')
+    global nvrtcCompileProgram
+    nvrtcCompileProgram = <F_nvrtcCompileProgram>_L.get('CompileProgram')
+    global nvrtcGetPTXSize
+    nvrtcGetPTXSize = <F_nvrtcGetPTXSize>_L.get('GetPTXSize' if _L.prefix == 'nvrtc' else 'GetCodeSize')  # NOQA
+    global nvrtcGetPTX
+    nvrtcGetPTX = <F_nvrtcGetPTX>_L.get('GetPTX' if _L.prefix == 'nvrtc' else 'GetCode')  # NOQA
+    global nvrtcGetCUBINSize
+    nvrtcGetCUBINSize = <F_nvrtcGetCUBINSize>_L.get('GetCUBINSize')
+    global nvrtcGetCUBIN
+    nvrtcGetCUBIN = <F_nvrtcGetCUBIN>_L.get('GetCUBIN')
+    global nvrtcGetProgramLogSize
+    nvrtcGetProgramLogSize = <F_nvrtcGetProgramLogSize>_L.get('GetProgramLogSize')  # NOQA
+    global nvrtcGetProgramLog
+    nvrtcGetProgramLog = <F_nvrtcGetProgramLog>_L.get('GetProgramLog')
+    global nvrtcAddNameExpression
+    nvrtcAddNameExpression = <F_nvrtcAddNameExpression>_L.get('AddNameExpression')  # NOQA
+    global nvrtcGetLoweredName
+    nvrtcGetLoweredName = <F_nvrtcGetLoweredName>_L.get('GetLoweredName')
+    global nvrtcGetNumSupportedArchs
+    nvrtcGetNumSupportedArchs = <F_nvrtcGetNumSupportedArchs>_L.get('GetNumSupportedArchs')  # NOQA
+    global nvrtcGetSupportedArchs
+    nvrtcGetSupportedArchs = <F_nvrtcGetSupportedArchs>_L.get('GetSupportedArchs')  # NOQA
+    global nvrtcGetNVVMSize
+    nvrtcGetNVVMSize = <F_nvrtcGetNVVMSize>_L.get('GetNVVMSize')
+    global nvrtcGetNVVM
+    nvrtcGetNVVM = <F_nvrtcGetNVVM>_L.get('GetNVVM')
+
+
+cdef SoftLink _get_softlink():
+    cdef int runtime_version
+    cdef str prefix = 'nvrtc'
+    cdef object libname = None
+
+    if CUPY_CUDA_VERSION != 0:
+        runtime_version = runtime.runtimeGetVersion()
+        if 11020 <= runtime_version < 12000:
+            # CUDA 11.x (11.2+)
+            if _sys.platform == 'linux':
+                libname = 'libnvrtc.so.11.2'
+            else:
+                libname = 'nvrtc64_112_0.dll'
+        elif 12000 <= runtime_version < 13000:
+            # CUDA 12.x
+            if _sys.platform == 'linux':
+                libname = 'libnvrtc.so.12'
+            else:
+                libname = 'nvrtc64_120_0.dll'
+    elif CUPY_HIP_VERSION != 0:
+        runtime_version = runtime.runtimeGetVersion()
```

Comment:
This needs an actual device when compiling. Is this behavior intended?
ROCm build only CI fails

Reply:
It's failing because ROCm build CI is doing `import cupy` but `runtime.runtimeGetVersion()` won't work without actual device. I'll fix.

Reply:
This should be a problem for CUDA 11.x as well...
https://github.com/cupy/cupy/issues/6688#issuecomment-1383216671

Reply:
This should be fixed.